go/oasis-node/cmd/common/metrics/resource.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/service"
+)
+
+// ResourceCollector is interface for monitoring resources (cpu, mem, disk, net).
+type ResourceCollector interface {
+	// Update updates corresponding resource metrics.
+	Update() error
+}
+
+// resourceService updates and collects number of resources.
+type resourceService struct {
+	service.BaseBackgroundService
+
+	interval           time.Duration
+	resourceCollectors []ResourceCollector
+}
+
+func (brs *resourceService) Start() error {
+	go brs.worker()
+	return nil
+}
+
+func (brs *resourceService) worker() {
+	t := time.NewTicker(brs.interval)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-brs.Quit():
+			break
```

Comment:
This will only break out of the select statement not the for loop. Use `return`.

Reply:
Ok. I also fixed it here https://github.com/oasislabs/oasis-core/blob/master/go/oasis-node/cmd/common/metrics/metrics.go#L169